core/src/main/java/org/jdbi/v3/core/argument/internal/PojoPropertyArguments.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.argument.internal;
+
+import java.util.Map;
+import java.util.Optional;
+
+import org.jdbi.v3.core.argument.NamedArgumentFinder;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.mapper.reflect.BeanMapper;
+import org.jdbi.v3.core.mapper.reflect.internal.PojoProperties.PojoProperty;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.core.statement.UnableToCreateStatementException;
+
+/**
+ * Inspect a object and bind parameters via {@link BeanMapper}'s properties.
+ */
+public class PojoPropertyArguments extends MethodReturnValueNamedArgumentFinder {
+    private final Map<String, ? extends PojoProperty<?>> properties;
+    private final StatementContext ctx;
+
+    /**
+     * @param prefix an optional prefix (we insert a '.' as a separator)
+     * @param bean the bean to inspect and bind
+     * @param ctx the statement context
+     */
+    public PojoPropertyArguments(String prefix, Object bean, StatementContext ctx) {
+        super(prefix, bean);
+        this.ctx = ctx;
+        final RowMapper<? extends Object> mapper = ctx.findRowMapperFor(bean.getClass())
+                .orElseThrow(() -> new UnableToCreateStatementException("Couldn't find registered property mapper for " + bean.getClass()));
+        if (!(mapper instanceof BeanMapper<?>)) {
+            throw new UnableToCreateStatementException("Registered mapper for " + bean.getClass() + " is not a property based mapper");
```

Comment:
What's the difference between PojoPropertyArguments and BeanPropertyArguments?

Reply:
`BeanPropertyArguments` is public API and in a different package, and as such it's really hard to extend it without accidentally introducing public API.  Even a new constructor shows up... will see if there's a better way to do this.

This probably also should have been `internal`.

Reply:
Cool, I fixed it by making the new stuff `protected`, and this is now simply a stub subclass (`stubclass` :trollface: ).  I marked `BeanPropertyArguments` deprecated.  We can clean it all up eventually.